### PR TITLE
Fix command runtime so StopUpstreamCommandsException doesn't get populated in `-ErrorVariable`

### LINKED ISSUE #9185: Select-Object -First populates `-ErrorVariable`
# Steps to reproduce

```powershell
& {
    [CmdletBinding()]param()
    1,2,3,4 | Select -First 1
} -ErrorVariable Problems; 
$Problems |  Format-List * -Force
```

# Expected behavior

There are no errors, so `$Problems` should be **empty**.

# Actual behavior

`$Problems` has `StopUpstreamCommandsException` which `Select-Object -First` threw.

# Environment data

```none
Name                           Value
----                           -----
PSVersion                      6.1.3
PSEdition                      Core
GitCommitId                    6.1.3
OS                             Microsoft Windows 10.0.18353
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

```

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
# PR Summary

`StopUpstreamCommandsException` is a special exception used by cmdlets to tell PowerShell to stop upstream commands as it doesn't need any additional input.  For example: "1,2,3 | select-object -first 1" should stop the first part of the pipeline after `select-object` received 1 object as the request has been fulfilled.  However, if you use `1,2,3 | select-object -first 1 -errorvariable err`, the internal exception (which is handled) is put into the `$err` variable (although `$error` it not populated).  This means that scripts may treat this as an error or special case this `System error` and throwing it away.

The fix is in the code where the engine populates `-ErrorVariable` to discard `StopUpstreamCommandsException`.

I don't believe this is a breaking change as anyone working around this by throwing away that specific error (because `select-object` actually succeeded) won't be impacted by this change.

## PR Context

Fix https://github.com/PowerShell/PowerShell/issues/9185

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: test/powershell/Modules/Microsoft.PowerShell.Utility/Select-Object.Tests.ps1 [hunk 1]
```diff
@@ -113,7 +113,12 @@ Describe "Select-Object" -Tags "CI" {
 	$result[0].Size              | Should -Be ($orig1 + 1)
 	$dirObject[0].Size           | Should -Be ($orig1 + 1)
 	$dirObject[$TestLength].Size | Should -Be ($orig2 + 1)
-    }
+	}
+
+	It "Should not leak internal exception for stopping upstream" {
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/Select-Object.Tests.ps1]: I think we should be more clear that the test checks:
```suggestion
	It "Should not leak 'StopUpstreamCommandsException' internal exception for stopping upstream" {
```

**Comment by iSazonov** [comment]
@SteveL-MSFT Why do you start to add prefixes to PR titles? This is contrary to our conventions.

**Comment by SteveL-MSFT** [comment]
@iSazonov trying something based on an article I read about good commit messages.  I suppose the prefix can be just in the commit and not in the PR title.

**Comment by iSazonov** [comment]
@SteveL-MSFT All that we can not formalize is a human factor. I mean, something that can make things clearer can also be misleading. In this case, the simpler the better. So better not to have prefixes at all. 
The most useful thing that we now have is the PR number in the commit title. This allows us to quickly find PR on GitHub and a history of the change. GitLens in VS Code greatly support this.

**Review by iSazonov — APPROVED** [review]

**Comment by vexx32** [comment]
@SteveL-MSFT `:bug:` (🐛) `-gt` `BUG:` 😉 

aka https://gitmoji.carloscuesta.me/

But regardless, I agree that for commit messages having an easily-identifiable prefix can be a nice-to-have. Whether that's emoji or text is, I suppose, up to you guys. 😄

**Review by anmenaga — APPROVED** [review]

**Pull request merged**